include/openthread/message.h

```diff
@@ -321,23 +321,17 @@ void otMessageQueueInit(otMessageQueue *aQueue);
  * @param[in]  aQueue    A pointer to the message queue.
  * @param[in]  aMessage  The message to add.
  *
- * @retval OT_ERROR_NONE     Successfully added the message to the queue.
- * @retval OT_ERROR_ALREADY  The message is already enqueued in a queue.
- *
  */
-otError otMessageQueueEnqueue(otMessageQueue *aQueue, otMessage *aMessage);
+void otMessageQueueEnqueue(otMessageQueue *aQueue, otMessage *aMessage);
```

Comment:
Thoughts about the approach from https://github.com/openthread/openthread/pull/4939#issuecomment-626108171 to keep the public API same (and do error handling in the `message_api.cpp`)?

Reply:
I see both sides. On one hand, it is desirable to keep public APIs stable. On the other hand, it was argued in https://github.com/openthread/openthread/pull/4939#issuecomment-626086485 that returning `OT_ERROR_ALREADY` leads to calling code handling errors incorrectly. Thoughts?

Reply:
I fully agree that handling error in higher layer gets tricky (we needed to handle for unicast/multicast address add/remove). I do also prefer that in such cases that we change both the private and  public APIs. 

But for the `MessageQueueEnqueue` case, I think it may be worth treating it differently:
- In other cases, like addresses or coap resource, ..., there is a single container/queue where the entries are being added/removed, so if user tries to add/remove same thing again, it should be fine to report `ERROR_NONE`.
- But for `Message` it can be in a different queue than the one user is calling `Enqueue` on, and I think we should consider this pattern a bug (incorrect use).
- Forgetting to dequeue a message and then trying to add it to another/same queue is a pattern that I feel can happen by mistake.

I would suggest for internal use we explicitly require that the `Message` passed to `Enqueue` to NOT be in any other queue and `OT_ASSERT` if it is (even if it is already in the same queue). 

For external API, I say we either provide an error in this case, or we can apply/require the same rule (I would favor we apply same rule to keep things simpler).

Thoughts?

Reply:
I generally prefer to be more strict as well. I updated this PR as suggested. Please review.